internal/services/paloalto/next_generation_firewall_vhub_local_rulestack_resource.go

```diff
@@ -276,6 +280,15 @@ func (r NextGenerationFirewallVHubLocalRuleStackResource) Update() sdk.ResourceF
 				firewall.Tags = tags.Expand(model.Tags)
 			}
 
+			if metadata.ResourceData.HasChange("rulestack_id") {
```

Comment:
should we be locking when `rulestack_id` has not changed too?

Reply:
TBH I'm not sure about this because it seems to be related to the API implementation. But intuitively, we only need lock when `rulestack_id` changes as it's only a reference to the rulestack resource.

Reply:
we could add the locks after line 251 then to avoid checking if it's changed and parsing it twice

Reply:
I have ever thought about it before, but now moved the locks after L251.